### PR TITLE
Add handy graph make targest, fix go-deps circular dependnecy, fix clean-input-srpms

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Add some handy `.PHONY` targets for graph files.
- `parsed-specs: $(specs_file)`
- `workplan graph: $(graph_file)`
- `graph-preprocessed: $(preprocessed_file)`

There was a harmless circular dependency in go tools on the new `go-deps` target. We now rely on `$(go_common_files)` to trigger it.

We didn't clean the toolchain srpms flag when running `clean-input-srpms` before.

###### Change Log  

- More graph targets
- Circular go deps fix
- `clean-input-srpms` now also removes toolchain srpm flag

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local

### CODE REVIEW

File: toolkit/scripts/safeunmount.sh [hunk 1]
```diff
@@ -2,23 +2,46 @@
...
-            sleep 1
-            if [ ${retries} -eq 0 ] ; then exit 1 ; fi
-        done
+    done || return 1
+
+    return 0
+}
+
+# For each argument, pass it to clean_dir in parallel then wait for all to finish and return the exit code
+for dir in "$@" ; do
```
Comment (neha170) [inline toolkit/scripts/safeunmount.sh:34]: Should we instantiate an empty array in the beginning
`pids=()
`

File: toolkit/scripts/safeunmount.sh [hunk 2]
```diff
@@ -2,23 +2,46 @@
...
         fi
-        retries=10
-        while mountpoint -q ${dir} ; do
-            echo "ERROR: Mountpoint still present at $dir, retrying unmount ${retries} times"
-            umount -l ${dir}
-            retries=$(( ${retries} - 1))
-            sleep 1
-            if [ ${retries} -eq 0 ] ; then exit 1 ; fi
-        done
+    done || return 1
```
Comment (neha170) [inline toolkit/scripts/safeunmount.sh:28]: Why do we have a `return 1` here? What errors, besides umount failure, can occur?

**Comment by neha170** [comment]
Could you add some description around:
- the circular dependency, and how you're resolving it
- reasons for new graph and graph-preprocessed targets
Are these changes related? It would probably be nicer to have 3 separate PRs if they're not related
Thanks :)

File: toolkit/scripts/pkggen.mk [hunk 3]
```diff
@@ -207,20 +210,23 @@ $(preprocessed_file): $(cached_file) $(go-graphPreprocessor)
...
 # Execute the package build scheduler.
 build-packages: $(RPMS_DIR)
 
 clean: clean-build-packages clean-compress-rpms clean-compress-srpms
-clean-build-packages:
+
+# Empty rule for clean-build-packages-workers, we will append more rules to this later via the $(eval) function below
+clean-build-packages-workers:
+	@echo Verifying no mountpoints present in $(CHROOT_DIR)
+	$(SCRIPTS_DIR)/safeunmount.sh "$(CHROOT_DIR)"/* && \
```
Comment (cwize1) [inline toolkit/scripts/pkggen.mk:223]: I'm guessing the `CHROOT_DIR` directory isn't ever a mountpoint itself, correct?

**Review by cwize1 — APPROVED** [review]

**Comment by dmcilvaney** [comment]
> Could you add some description around:
> 
> * the circular dependency, and how you're resolving it
> * reasons for new graph and graph-preprocessed targets
>   Are these changes related? It would probably be nicer to have 3 separate PRs if they're not related
>   Thanks :)

Sure, I can split these up a bit. There were bouncing around my local branches for months, needed to get them up at some point.

I'll probably split the chroot cleanup into its own PR.

File: toolkit/scripts/pkggen.mk (hunk 3 above)
Comment (dmcilvaney) [reply toolkit/scripts/pkggen.mk:223]: Its case-by-case. Here each package will have its own unique chroot so the /* makes sense.

File: toolkit/scripts/safeunmount.sh (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/scripts/safeunmount.sh:34]: Sure, I can add that for more clarity.

File: toolkit/scripts/safeunmount.sh (hunk 2 above)
Comment (dmcilvaney) [reply toolkit/scripts/safeunmount.sh:28]: The loop itself can error out if the input is garbage, and I've found error handling of background tasks unpredictable at the best of times. I tend to try and be explicit with return codes as much as possible. Maybe another approach would be to assume `return 1`, and only `return 0` if a final call to `mountpoint -q` says we are unmounted.

File: toolkit/scripts/safeunmount.sh (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/scripts/safeunmount.sh:34]:
```suggestion
pids=()
for dir in "$@" ; do
```

**Comment by dmcilvaney** [comment]
Ok, removed unmount and merged latest main

**Comment by dmcilvaney** [comment]
https://github.com/microsoft/CBL-Mariner/pull/6262

**Comment by dmcilvaney** [comment]
Added the new doc strings with the merge.

File: toolkit/scripts/safeunmount.sh (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/scripts/safeunmount.sh]: Moved to https://github.com/microsoft/CBL-Mariner/pull/6262

File: toolkit/scripts/safeunmount.sh (hunk 2 above)
Comment (dmcilvaney) [reply toolkit/scripts/safeunmount.sh]: Moved to https://github.com/microsoft/CBL-Mariner/pull/6262

File: toolkit/scripts/pkggen.mk [hunk 4]
```diff
@@ -43,10 +43,15 @@ logging_command = --log-file=$(LOGS_DIR)/pkggen/workplan/$(notdir $@).log --log-
 $(call create_folder,$(LOGS_DIR)/pkggen/workplan)
 $(call create_folder,$(rpmbuilding_logs_dir))
 
-.PHONY: clean-workplan clean-cache clean-cache-worker clean-grapher-cache-worker clean-spec-parse clean-ccache graph-cache analyze-built-graph workplan
+.PHONY: clean-workplan clean-cache clean-cache-worker clean-grapher-cache-worker clean-spec-parse clean-ccache graph graph-cache graph-preprocessed analyze-built-graph workplan
+##help:target:parsed-specs=Parse package specs and generate a specs.json file encoding all dependency information.
+parsed-specs: $(specs_file)
```
Comment (neha170) [inline toolkit/scripts/pkggen.mk:48]: nit: do we want to rename this to parse-specs, as other targets sound like 'this target _wil_ ..' and this one sounds like 'this target _has_ ..'.

File: toolkit/scripts/pkggen.mk [hunk 5]
```diff
@@ -43,10 +43,15 @@ logging_command = --log-file=$(LOGS_DIR)/pkggen/workplan/$(notdir $@).log --log-
...
+##help:target:graph-cache=Resolve package dependencies and cache the results.
 graph-cache: $(cached_file)
-##help:target:workplan=Create the package build workplan.
-workplan: $(graph_file)
+##help:target:graph=Create the initial package build graph.
+workplan graph: $(graph_file)
```
Comment (neha170) [inline toolkit/scripts/pkggen.mk:52]: do we plan to get rid of 'workplan' target later?

**Review by neha170 — APPROVED** [review]
Thanks for providing the context Daniel. LGTM. Minor suggestions, not blocking

File: toolkit/scripts/pkggen.mk (hunk 4 above)
Comment (dmcilvaney) [reply toolkit/scripts/pkggen.mk:48]: Sure, that seems reasonable.
```suggestion
parse-specs: $(specs_file)
```

File: toolkit/scripts/pkggen.mk (hunk 5 above)
Comment (dmcilvaney) [reply toolkit/scripts/pkggen.mk:52]: No, I expect it will have to stay at least until 3.0. I don't want to break any existing workflows.

**Pull request merged**